hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMORDataSource.scala

```diff
@@ -278,7 +278,6 @@ class TestMORDataSource extends HoodieClientTestBase {
     val inputDF5: Dataset[Row] = spark.read.json(spark.sparkContext.parallelize(records5, 2))
     inputDF5.write.format("org.apache.hudi")
       .options(commonOpts)
-      .option("hoodie.compact.inline", "true")
```

Comment:
@garyli1019 Please also help to review, i delete the hoodie.compact.inline = true . Because when overwrite use replace commit, "Fifth Operation: " compaction will postponed until "Sixth Operation:", so " assertEquals(152, hudiIncDF6.count())" will be ok. 
But when i modify the overwrite to fs.delete, compaction will do in "Fifth Operation: ", so after commit5Time will not have compaction commit, hudiIncDF6.count() will be 2. 
So delete the inline compaction in "Fifth Operation: ", and it will do in "Sixth Operation:"
https://github.com/apache/hudi/pull/1938/files

Reply:
Thanks @lw309637554 , this compaction was not supposed to be here. This probably the cause of the flaky test as well. cc: @vinothchandar

Reply:
This is great. Thanks for diggin in both @lw309637554 !